aether-kernel/sql/create_readonly_user.py

```diff
@@ -59,6 +59,7 @@
 GRANT SELECT ON kernel_mapping TO {role_id};
 GRANT SELECT ON kernel_schemadecorator TO {role_id};
 GRANT SELECT ON kernel_schema TO {role_id};
+GRANT SELECT ON multitenancy_mtinstance TO {role_id};
```

Comment:
The problem with this is that if multi-tenancy is not enabled, this table does not exists at all :disappointed:

Reply:
Oh no. Looks like I need a mode switch for the producer. How can we make the readonlyuser creation reactive to the system config?

Reply:
Ask to the _DB expert_ :smiling_imp:

Reply:
If the table does not exits... create a view to simulate it  _expert says_

```sql
GRANT SELECT ON kernel_project TO {role_id};

DROP VIEW IF EXISTS multitenancy_mtinstance;
DO $$
BEGIN
  IF NOT EXISTS (SELECT FROM information_schema.tables WHERE table_name = 'multitenancy_mtinstance')
  THEN
      CREATE VIEW multitenancy_mtinstance AS
            SELECT '-'    AS realm,
                    p.id  AS instance_id
            FROM kernel_project AS p;
  END IF;
END
$$ LANGUAGE plpgsql;
```